src/app/src/main/java/com/cmput301f21t09/budgetprojectname/HabitEventModel.java

```diff
@@ -9,55 +9,181 @@
 import java.io.Serializable;
 import java.util.Date;
 
+/**
+ * Represents the habit event model containing fields like habit event ID, location, comment, date,
+ * image, and habit ID
+ */
 public class HabitEventModel{
-    private final String ID; // TODO: change type depending on Firebase ID type
-    private final String location; // TODO: change to Location once this is implemented
-    private final String description;
-    private final Date date;
-    private final Image image;
+    /**
+     * HabitEvent ID
+     */
+    private String ID;
 
-    HabitEventModel(String location, Date date, String description, Image image) {
+    /**
+     * Location of habit event completion (optional)
+     */
+    private String location; // TODO: change to Location type once this is implemented
+
+    /**
+     * Comment of habit event (optional)
+     */
+    private String comment;
+
+    /**
+     * Date of habit event completion
+     */
+    private Date date;
+
+    /**
+     * Image of habit event (optional)
+     */
+    private Image image;
+
+    /**
+     * Habit ID to reference completed habit
+     */
+    private String habitID;
+
+    HabitEventModel(String location, Date date, String comment, Image image, String habitID) {
         this.ID = "s1w9kwequ290"; // placeholder ID
         this.location = location;
-        this.description = description;
+        this.comment = comment;
         this.date = date;
         this.image = image;
+        this.habitID = habitID;
     }
 
-    HabitEventModel(String location, Date date, String description) {
+    HabitEventModel(String location, Date date, String comment, String habitID) {
```

Comment:
Add documentation for constructors please

Reply:
Added and removed unused contstructors.